### ISSUE TITLE
How can I update the license, copyright, and package information that Scancode uses to scan for in the installation directory?

### ISSUE THREAD

**Issue by KimDongMinWiKi** [nexB/scancode-toolkit#3394]
Hello, brothers and sisters. I am studying Scancode and I want to add additional licenses that are continually being added to the directory where Scancode is installed. However, I am not sure how to proceed. What should I do?

**Comment by AyanSinhaMahapatra** [comment]
@KimDongMinWiKi 
See for licenses: 
* https://scancode-toolkit.readthedocs.io/en/latest/how-to-guides/add_new_license.html
* https://scancode-toolkit.readthedocs.io/en/latest/how-to-guides/add_new_license_detection_rule.html
* https://scancode-toolkit.readthedocs.io/en/latest/how-to-guides/install_new_license_plugin.html

For copyrights we use pattern matching like here: https://github.com/nexB/scancode-toolkit/blob/develop/src/cluecode/copyrights.py

And for packages we have parsing and assembling for different types of package manifest here:https://github.com/nexB/scancode-toolkit/tree/develop/src/packagedcode

**Comment by KimDongMinWiKi** [comment]
@AyanSinhaMahapatra 

In which subdirectory should the new license file be placed in the released directory?

Or are you building from source and using that?

**Comment by MihaiChirciu** [comment]
@AyanSinhaMahapatra Hi,

I am new to Scancode and I did not manage to add a new license or rule. I have installed Scancode (version 31.2.4) on a windows 10 machine, via Git Bash by cloning the github project. I ran the tool on some test files and it seems to be working fine, however I wanted add a new license and a new rule for it.

- I have created: \src\licensedcode\data\licenses\zz-test.LICENSE with the following content:

---
key: mihai-license
short_name: Mihai License
name: Mihai License
owner: Mihai

---

This is my test license: Mihai License

This is my custom test license.
Use for testing purposes only!

- Then I have created: \src\licensedcode\data\rules\zz-test.RULE with the following content: (after seeing that is not working, I have created zz-test_1.RULE where I have the plain text without "* " in front of it; it still did not work).

---
license_expression: mihai-license
is_license_text: yes
minimum_coverage: 80
---

* This is my custom test license.
* Use for testing purposes only!

- Then I scanned the following file mihai_file.h that contains:

/*
 *  Copyright Mihai C.
 *
 *  This is my custom test license.
 *  Use for testing purposes only!
 */

#ifndef _SAMPLE_DRV_H_
#define _SAMPLE_DRV_H_


I was expecting the tool to detect mihai-license, but it doesn't. I have read the documentation and it doesn't say I should do anything else. I even tried the old format with RULE or LICENSE plus the yaml file.
I am new to this, so I have also ran ./configure maybe it would help or anything (but it didn't).

Is there something that I am missing? Is there an extra step that I need to take after creating a license or a rule in order to "notify" the tool about it?

**Comment by AyanSinhaMahapatra** [comment]
@MihaiChirciu 

See the `note` section on the doc I mentioned before: https://scancode-toolkit.readthedocs.io/en/latest/how-to-guides/add_new_license.html

> Add licenses in a local development installation and run `scancode-reindex-licenses` to make sure we reindex the licenses and this validates the new licenses.

This should be run inside the configured virtualenv that you are creating, of course.

Can you confirm you are running this step too?

**Comment by AyanSinhaMahapatra** [comment]
@KimDongMinWiKi 

> Or are you building from source and using that?

Yes! You should be doing this in a git checkout if you are adding one license file, in the original location that these license files exist.

You can also work from a downlaoded release archive, but then you have to install a plugin like the doc mentions here: https://scancode-toolkit.readthedocs.io/en/latest/how-to-guides/install_new_license_plugin.html

But this is more advanced use case.

> In which subdirectory should the new license file be placed in the released directory?

From the docs: https://scancode-toolkit.readthedocs.io/en/latest/how-to-guides/add_new_license.html

> All licenses are stored as a plain text file in the src/licensedcode/data/licenses directory using their key as base for the file name. For example the filename for a license with key: mit would be mit.LICENSE.

**Comment by pombredanne** [comment]
@MihaiChirciu you may want to start a separate issue to avoid mixing things up with @KimDongMinWiKi threads who started this issue? This would be very kind of you! Note BTW that if you create a `zz-test.LICENSE` file, the KEY must be the same as base file name, e.g., `key: zz-test` should be in the YAML. Also you need to run the license reindexing command once this is done.

**Comment by MihaiChirciu** [comment]
@AyanSinhaMahapatra I did not run scancode-reindex-licenses.
@pombredanne I changed the key to "zz-test", then I ran scancode-reindex-licenses, ran the scan again and still got only the copyright in the report. I will open another issue in order not to diverge from the main thread.
Thank you for your response and guidance so far.
